src/disk_io_thread.cpp

```diff
@@ -1087,10 +1104,6 @@ namespace libtorrent
 		TORRENT_ASSERT(j->next == 0);
 		TORRENT_ASSERT((j->flags & disk_io_job::in_progress) || !j->storage);
 
-		mutex::scoped_lock l(m_cache_mutex);
-
-		check_cache_level(l, completed_jobs);
-
 		DLOG("perform_job job: %s ( %s%s) piece: %d offset: %d outstanding: %d\n"
```

Comment:
@ssiloti:  Just keep in mind that some people actually do build Libtorrent as 64-bit (I actually prefer such builds), so try not to break that if you can help it.